circuits/plonk-5-wires/src/scalars.rs

```diff
@@ -112,3 +61,143 @@ impl<F: Field> RandomOracles<F> {
         }
     }
 }
+
+//
+// OCaml types
+//
+
+#[cfg(feature = "ocaml_types")]
+pub mod caml {
+    use super::*;
+    use oracle::sponge::caml::CamlScalarChallenge;
+
+    //
+    // ProofEvaluations<F> <-> CamlProofEvaluations<CamlF>
+    //
+
+    #[derive(Clone, ocaml::ToValue, ocaml::FromValue)]
+    pub struct CamlProofEvaluations<F> {
+        pub w: (Vec<F>, Vec<F>, Vec<F>, Vec<F>, Vec<F>),
+        pub z: Vec<F>,
+        pub t: Vec<F>,
+        pub f: Vec<F>,
+        pub s: (Vec<F>, Vec<F>, Vec<F>, Vec<F>),
+    }
+
+    impl<F, CamlF> From<ProofEvaluations<Vec<F>>> for CamlProofEvaluations<CamlF>
+    where
+        F: Clone,
+        CamlF: From<F>,
+    {
+        fn from(pe: ProofEvaluations<Vec<F>>) -> Self {
+            let w = (
+                pe.w[0].iter().cloned().map(Into::into).collect(),
+                pe.w[1].iter().cloned().map(Into::into).collect(),
+                pe.w[2].iter().cloned().map(Into::into).collect(),
+                pe.w[3].iter().cloned().map(Into::into).collect(),
+                pe.w[4].iter().cloned().map(Into::into).collect(),
```

Comment:
This looks like 5-wire stuff. I thought we are getting rid of it?

Reply:
yeah, in the 15-wires branch it is removed :o